src/schema/firefox-schemas-import.js

```diff
@@ -473,47 +482,88 @@ export function foldSchemas(schemas) {
   return foldedSchemas;
 }
 
+// Normalize the JSONSchema, gets an array of JSONSchema objects and return an array of
+// normalized JSONSchema objects (a JSONSchema file may contain more than one API namespaces,
+// e.g. like menus.json and browser_action.json).
+//
+// (schemas: Array<Object>, file: string) -> Array<Object>
 inner.normalizeSchema = (schemas, file) => {
   const filteredSchemas = foldSchemas(filterSchemas(schemas));
-  let extendSchemas;
-  let primarySchema;
+  let manifestExtendSchemas;
+  let apiNamespaceSchemas;
 
   if (filteredSchemas.length === 1) {
     // If there is only a manifest namespace then this just extends the manifest.
     if (
       filteredSchemas[0].namespace === 'manifest' &&
       file !== 'manifest.json'
     ) {
-      primarySchema = {
-        namespace: file.slice(0, file.indexOf('.')),
-      };
-      extendSchemas = [filteredSchemas[0]];
+      apiNamespaceSchemas = [
+        {
+          namespace: file.slice(0, file.indexOf('.')),
+        },
+      ];
+      manifestExtendSchemas = [filteredSchemas[0]];
     } else {
-      [primarySchema] = filteredSchemas;
-      extendSchemas = [];
+      apiNamespaceSchemas = filteredSchemas;
+      manifestExtendSchemas = [];
     }
   } else {
-    extendSchemas = filteredSchemas.slice(0, filteredSchemas.length - 1);
-    primarySchema = filteredSchemas[filteredSchemas.length - 1];
+    manifestExtendSchemas = filteredSchemas.filter((s) => {
+      return s.namespace === 'manifest' && file !== 'manifest.json';
+    });
+    apiNamespaceSchemas = filteredSchemas.filter((s) => {
+      return !manifestExtendSchemas.includes(s);
+    });
   }
 
-  const { namespace, types, ...rest } = primarySchema;
-  const { types: extendTypes, ...extendRest } = rewriteExtend(
-    extendSchemas,
-    namespace
-  );
-  const updatedTypes = { ...loadTypes(types), ...extendTypes };
-  return {
-    ...rest,
-    ...extendRest,
-    id: namespace,
-    types: updatedTypes,
-  };
+  // Rewrite manifest extend types using the each api namespace in the file.
+  return apiNamespaceSchemas.map((apiScheme) => {
+    let { namespace, types, ...rest } = apiScheme; // eslint-disable-line prefer-const
```

Comment:
If you extract `$import` here, do you need to (1) use `let` and (2) `delete` it later?

Reply:
You could potentially rename `namespace` to `id` here so that you don't need to do it at the end.

Reply:
I'll take another look, I wouldn't exclude I can replace this eslint-disable-line with a few tweaks to the code instead (honestly I just rubberstamped the eslint-disable-line because I wanted to be sure that everything was working before spending some more time tweaking this function to avoid that, but I was just being a bit "lazy"... errr "efficient" :-P)

Reply:
@willdurand ah, in my comment with a summary of the last round of changes I forgot to mention that I removed this `eslint-disable-line` suppress comment, with just a couple of some small tweaks to this arrow function.